controllers/openstackmachine_controller.go

```diff
@@ -348,6 +348,12 @@ func (r *OpenStackMachineReconciler) reconcileNormal(ctx context.Context, scope
 	}
 
 	addresses := instanceNS.Addresses()
+
+	// Append the server's name as the NodeInternalDNS to match the node name of a CSR created for a node that used in-tree provider.
```

Comment:
Can we expand on this comment a bit? This makes it sound like we might be able to remove this now. However, we can probably never remove this because it's required for nodes which were upgraded from in-tree.

Reply:
Done